File: libraryCoreJava/src/main/java/com/juqitech/moretickets/core/base/impl/BaseApp.java
package com.juqitech.moretickets.core.base.impl;

import android.app.Application;

/**
 * @author zhanfeng
 * @date 2019-06-05
 * @desc
 */
public class BaseApp extends Application {

    private static BaseApp instance;


    public static synchronized BaseApp getApplication() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }
}

File: libraryCore/src/main/java/com/juqitech/moretickets/core/base/impl/BaseApp.kt
package com.juqitech.moretickets.core.base.impl

import android.app.Application
import kotlin.properties.Delegates

/**
 * @author zhanfeng
 * @date 2019-06-05
 * @desc
 */
open class BaseApp : Application() {

    override fun onCreate() {
        super.onCreate()
        application = this
    }

    companion object {

        @get:Synchronized
        var application: BaseApp by Delegates.notNull()
            private set
    }
}
